libsolidity/codegen/MultiUseYulFunctionCollector.h

```diff
@@ -54,6 +55,8 @@ class MultiUseYulFunctionCollector
 private:
 	/// Map from function name to code for a multi-use function.
 	std::map<std::string, std::string> m_requestedFunctions;
+
+	std::set<std::string> m_beingGeneratedFunctions;
```

Comment:
I added this as some functions, for example in case of recursive struct end up being recursions at the end...

If this is fine, I should document this field :)